### DISCUSSION TITLE
How can I change what the "image" field type saves in frontmatter while still allowing the CMS image thumbnail to work?

### DISCUSSION THREAD

**Discussion by Boston343** [tinacms/tinacms#3917] [Help]
I'm working with Astro and Tina CMS. When working locally testing and I upload files with the below info, the text that gets put in the frontmatter is "/public/assets/images/img-name.jpg" which is great because it allows me to use Astro's built in image optimization. I could put it anywhere really so that part doesn't matter (Astro recommends `src/assets/`).

```ts
media: {
  tina: {
    publicFolder: "public",
    mediaRoot: "assets/images",
  },
},
```

When using the Tina Cloud on the deployed site and I insert an image it tries to insert it into the frontmatter as `https://assets.tina.io/a-bunch-of-numbers-and-letters/img-name.jpg`. I'm not sure why this happens since the file gets uploaded to the github repo anyway, and it does not allow me to use the image optimization. The only way so far I've found to change that is with the below:

```ts
{
  type: "image",
  name: "mainImage",
  label: "Main Image",
  required: true,
  ui: {
    parse(value) {
      // get everything after the last slash, looks like "img-name.jpg"
      const lastIdxOfSlash = value.lastIndexOf("/");

      // if no "/", return entire string
      if (lastIdxOfSlash === -1) {
        return value;
      }

      // change the frontmatter saved to work with astro image optimization
      return (
        "/public/assets/images/" + value.substring(lastIdxOfSlash + 1)
      );
    },
  },
},
```

This works and changes what gets put in the frontmatter, but when in Tina CMS after I select the image it shows up as a broken link. This isn't the end of the world as it still deploys correctly, but is there a way I can get it to still display the thumbnail while changing what gets put in the frontmatter? 

![image](https://user-images.githubusercontent.com/96832297/236701653-00001f38-2308-4d55-9dbe-68242a566b14.png)

**Comment by logan-anderson** [comment]
@Boston343 Are you using [the TinaCMS client for data fetching](https://tina.io/docs/features/data-fetching/#making-requests-with-the-tina-client) or just using the markdown files to build the site?

If you **are not** using tinacms (the generated client) for data fetching you can remove the parse function and `assets.tina.io` image URL will only be used when editing in the admin.

The reason you get the `assets.tina.io` domain is because when you upload an image you would have to wait until your site is redeployed to see it in the CMS. `assets.tina.io` is used whenever you use are data fetching and are using TinaCloud. This allows SSR and ISR to work much better. 

If you are using the tina client for data fetching will have a feature soon (See [this pr](https://github.com/tinacms/tinacms/pull/3887) ) that will allow you to build statically with the local image paths.

**Reply by Boston343** [reply]
@logan-anderson I'm currently using Astro's content collections rather than the TinaCMS client for data fetching. So I just want the image to be put in front matter as `/public/assets/images/img-name.jpg`. I'm pretty sure it was editing the github markdown file to be `https://assets.tina.io/a-bunch-of-numbers-and-letters/img-name.jpg` instead which was undesired. I'll try removing the parse later and double check it again to be sure, then report back.

**Reply by Boston343** [reply]
@logan-anderson okay so I see it's not putting things in with `https://assets.tina.io`, but I can't get it to put the word `public` in there. Unfortunately I need that in there to work with astro content collection image optimization (it needs full path). 

Goal: when I add an image it gets placed at `public/assets/images`, and it gets put in frontmatter as `/public/assets/images/img-name.jpg`.

So lets say I use the below config which places the image at the correct location:

```tsx
media: {
  tina: {
    publicFolder: "public",
    mediaRoot: "assets/images",
  },
},
```

However in the frontmatter this puts `/assets/images/img-name.jpg` when I need it to be `/public/assets/images/img-name.jpg`. So I can add the below parse in the original post to get that to work, but then the CMS image preview doesn't work. I figured parse would be the correct thing since it's stated to change the value put into the markdown, but not shown in Tina.

If I use the below config, it shows me the image preview in Tina correctly, but doesn't upload the image to the github repo correctly (uploads to public/public/assets/images)

```tsx
media: {
  tina: {
    publicFolder: "",
    mediaRoot: "/public/assets/images",
  },
},
```

I'm clearly trying to trick Tina into doing something it doesn't want to do. Is there a clean way to do this?
